packages/idempotency/src/persistence/BasePersistenceLayer.ts

```diff
@@ -0,0 +1,247 @@
+import {
+  createHash,
+  Hash
+} from 'node:crypto';
+import { search } from 'jmespath';
+import { IdempotencyRecordStatus } from '../types';
+import type {
+  BasePersistenceLayerOptions
+} from '../types';
+import { EnvironmentVariablesService } from '../config';
+import { IdempotencyRecord } from './IdempotencyRecord';
+import { BasePersistenceLayerInterface } from './BasePersistenceLayerInterface';
+import { IdempotencyValidationError } from '../Exceptions';
+
+abstract class BasePersistenceLayer implements BasePersistenceLayerInterface {
+  public idempotencyKeyPrefix: string;
+  private configured: boolean = false;
+  // envVarsService is always initialized in the constructor
+  private envVarsService!: EnvironmentVariablesService;
+  private eventKeyJmesPath?: string;
+  private expiresAfterSeconds: number = 60 * 60; // 1 hour default
+  private hashFunction: string = 'md5';
+  private payloadValidationEnabled: boolean = false;
+  private throwOnNoIdempotencyKey: boolean = false;
+  private useLocalCache: boolean = false;
+  private validationKeyJmesPath?: string;
+
+  public constructor() { 
+    this.setEnvVarsService();
+    this.idempotencyKeyPrefix = this.getEnvVarsService().getFunctionName();
+  }
+
+  /**
+   * Initialize the base persistence layer from the configuration settings
+   * 
+   * @param {BasePersistenceLayerConfigureOptions} config - configuration object for the persistence layer
+   */
+  public configure(config: BasePersistenceLayerOptions): void {
+    const { config: idempotencyConfig } = config;
+
+    if (config?.functionName && config.functionName.trim() !== '') {
```

Comment:
I ended up reverting this as we want to check whether it's defined & when defined whether it's a non-empty string. By using the notation you suggested, when not passing a `functionName` it would evaluate to `undefined`, making the `if` statement evaluate to `true` as it's different from empty string.